pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -423,44 +423,22 @@ func buildGatewayListenerTLSContext(
 		},
 	}
 
-	if server.Tls.CredentialName != "" {
-		// If SDS is enabled at gateway, and credential name is specified at gateway config, create
-		// SDS config for gateway to fetch key/cert at gateway agent.
+	switch {
+	// If SDS is enabled at gateway, and credential name is specified at gateway config, create
+	// SDS config for gateway to fetch key/cert at gateway agent.
+	case server.Tls.CredentialName != "":
 		authn_model.ApplyCustomSDSToServerCommonTLSContext(ctx.CommonTlsContext, server.Tls, authn_model.GatewaySdsUdsPath, requestedType)
-	} else if server.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL {
+	case server.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL:
 		authn_model.ApplyToCommonTLSContext(ctx.CommonTlsContext, metadata, sdsPath, server.Tls.SubjectAltNames, requestedType)
-	} else {
-		// Fall back to the read-from-file approach when SDS is not enabled or Tls.CredentialName is not specified.
-		ctx.CommonTlsContext.TlsCertificates = []*tls.TlsCertificate{
-			{
-				CertificateChain: &core.DataSource{
-					Specifier: &core.DataSource_Filename{
-						Filename: server.Tls.ServerCertificate,
-					},
-				},
-				PrivateKey: &core.DataSource{
-					Specifier: &core.DataSource_Filename{
-						Filename: server.Tls.PrivateKey,
-					},
-				},
-			},
-		}
-		var trustedCa *core.DataSource
-		if len(server.Tls.CaCertificates) != 0 {
-			trustedCa = &core.DataSource{
-				Specifier: &core.DataSource_Filename{
-					Filename: server.Tls.CaCertificates,
-				},
-			}
-		}
-		if trustedCa != nil || len(server.Tls.SubjectAltNames) > 0 {
-			ctx.CommonTlsContext.ValidationContextType = &tls.CommonTlsContext_ValidationContext{
-				ValidationContext: &tls.CertificateValidationContext{
-					TrustedCa:            trustedCa,
-					MatchSubjectAltNames: util.StringToExactMatch(server.Tls.SubjectAltNames),
-				},
-			}
+	case server.Tls.Mode == networking.ServerTLSSettings_MUTUAL || server.Tls.Mode == networking.ServerTLSSettings_SIMPLE:
```

Comment:
Add a default branch

Reply:
It seems this branch is not equal to previous

Reply:
ok. changed the case to default